kolibri/plugins/coach/assets/src/views/exams-page/preview-exam-modal.vue

```diff
@@ -11,36 +44,99 @@
 <script>
 
   const examActions = require('../../state/actions/exam');
+  const ContentNodeResource = require('kolibri').resources.ContentNodeResource;
+  const { createQuestionList, selectQuestionFromExercise } = require('kolibri.utils.exams');
 
   module.exports = {
     $trNameSpace: 'previewExamModal',
     $trs: {
-      preview: 'Preview',
+      preview: 'Preview Exam Exercises',
       close: 'Close',
+      question: 'Question { num }',
+      numQuestions: '{ num } questions',
+      randomize: 'Randomize',
+      exercise: 'Exercise { num }',
     },
     components: {
       'core-modal': require('kolibri.coreVue.components.coreModal'),
       'icon-button': require('kolibri.coreVue.components.iconButton'),
+      'content-renderer': require('kolibri.coreVue.components.contentRenderer'),
+      'ui-collapsible': require('keen-ui/src/UiCollapsible'),
     },
     props: {
-      examId: {
-        type: String,
-        required: true,
-      },
-      examTitle: {
-        type: String,
-        required: true,
-      },
-      classId: {
-        type: String,
+      exam: {
+        type: Object,
         required: true,
       },
     },
     methods: {
       close() {
         this.displayModal(false);
       },
+      isSelected(index) {
+        if (this.questionNumber === index) {
+          return 'selected';
+        }
+        return null;
+      },
+      goToQuestion(index) {
+        this.questionNumber = index;
+      },
     },
+    computed: {
+      seed() {
+        return this.exam.seed;
+      },
+      questionSources() {
+        try {
+          return JSON.parse(this.exam.question_sources);
+        } catch (e) {
+          if (e instanceof SyntaxError) {
+            return [];
+          }
+          throw e;
+        }
+      },
+      questions() {
+        return Object.keys(this.contentNodeMap).length ? createQuestionList(
+          this.questionSources).map(
+            question => ({
+              itemId: selectQuestionFromExercise(
+              question.assessmentItemIndex,
+              this.seed,
+              this.contentNodeMap[question.contentId]),
+              contentId: question.contentId
+            })
+        ) : [];
+      },
+      currentQuestion() {
+        return this.questions[this.questionNumber] || {};
+      },
+      content() {
+        return this.contentNodeMap[this.currentQuestion.contentId];
+      },
+      itemId() {
+        return this.currentQuestion.itemId;
+      },
+    },
+    created() {
+      ContentNodeResource.getCollection(
+        { channel_id: this.exam.channel_id },
+        { ids: this.questionSources.map(item => item.exercise_id) }
+        ).fetch().then(contentNodes => {
+          contentNodes.forEach(node => { this.$set(this.contentNodeMap, node.pk, node); });
+        });
+    },
```

Comment:
somewhat weird (to my eyes but apparently not eslint) whitespace in here, `questions()`, and a few other places in the PR

(Generally I'd expect to see parens and other brackets begin and end at the same level of indentation.)

Perhaps prettify will solve all our subjective consistency issues :)